src/store/modules/gateway/actions.ts

```diff
@@ -0,0 +1,59 @@
+import { Commit, ActionPayload } from 'vuex';
+import * as mutation from './types-mutations';
+import * as consts from '../../../consts/general';
+import {
+  GatewayState,
+  RootState,
+  AppGatewayServiceRequest,
+  AppGatewayServiceClient,
+  grpc,
+} from '../../types';
+
+
+interface ActionContext {
+  state: GatewayState;
+  rootState: RootState;
+  commit: Commit;
+  payload?: ActionPayload;
+}
+
+export const setNewClient = ({ state, commit }: ActionContext): any => {
+  const hostname: string = process.env.VUE_APP_HOST_NAME == null ? '' : process.env.VUE_APP_HOST_NAME;
+
+  if (hostname.length === 0) {
+    // TODO route to error 50X page
+    console.error('setNewClient: VUE_APP_HOST_NAME not loaded');
+    return;
+  }
+
+  const client: AppGatewayServiceClient = new AppGatewayServiceClient(hostname, null, null);
+  commit(mutation.SET_GRPC_CLIENT, client);
+};
+
+export const initAuthHeader = ({ state, commit }: ActionContext): any => {
+  let token: string | null = window.localStorage.getItem(consts.LOCAL_STORAGE_TOKEN_KEY);
+  let authType: string = consts.USER_AUTH;
+
+  if (token == null) {
+    const dummyEmail: string | undefined = process.env.VUE_APP_DUMMY_EMAIL;
+    const dummyPassword: string | undefined = process.env.VUE_APP_DUMMY_PASSWORD;
+    if (dummyEmail == null && dummyPassword == null) {
```

Comment:
Did you mean `dummyEmail == null || dummyPassword == null`?

Reply:
yes